### ISSUE TITLE
ClipAsync doesn't seem to work

### ISSUE THREAD

**Issue by bfmsoft** [excubo-ag/Blazor.Canvas#148]
I can't seem to get Clipping working.  I was wondering if I am missing something?  Below is the code

            await Context2D.BeginPathAsync();
            await Context2D.RectAsync(20, 20, 100, 100);
            await Context2D.ClipAsync(FillRule.NonZero);

            // I would expect this to NOT be show (outside the clipped part) but it is
            await Context2D.FillTextAsync("aaa", 25, 0);

Basically it seems the Clip is being ignored.  Can't seem to see any errors.  I looked through the sample and can't find any usage.  Any help would be create.

**Comment by stefanloerwald** [comment]
Hi @bfmsoft,

I can't see any difference between 
```cs
await Context2D.BeginPathAsync();
await Context2D.RectAsync(20, 20, 100, 100);
//await Context2D.ClipAsync(FillRule.NonZero);
await Context2D.FillTextAsync("aaa", 25, 0);
```
and
```cs
await Context2D.BeginPathAsync();
await Context2D.RectAsync(20, 20, 100, 100);
await Context2D.ClipAsync(FillRule.NonZero);
await Context2D.FillTextAsync("aaa", 25, 0);
```

In both cases I see nothing on the canvas. Could you please double check whether the code you showed is the code you experienced the issue with?

Thanks
Stefan

**Comment by bfmsoft** [comment]
I wrote up a sample project that shows what I mean.  I think it should work?

[WebApplication4.zip](https://github.com/excubo-ag/Blazor.Canvas/files/6517826/WebApplication4.zip)

**Comment by stefanloerwald** [comment]
You seem to have a misunderstanding of how the canvas API works. You can verify that this library works correctly, by executing the following javascript code:
```js
ctx.fillStyle = "black";
ctx.fillRect(0, 0, 400, 400);

ctx.beginPath();
ctx.rect(20, 25, 100, 100);
ctx.clip = "nonzero";

ctx.textAlign = "left";
ctx.fillStyle = "#FFFFFF";
ctx.font = "14px Arial";
ctx.fillText("This should be INvisible (outside of clipped region)", 25, 15);
ctx.fillText("This should be visible (because in clipped region)", 25, 40);
```

This is exactly equivalent to what's happening in your C# code and it shows the _exact_ same result. Therefore, your expectation of the text being invisible is simply incorrect.

Here's the documentation: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clip

**Comment by stefanloerwald** [comment]
My apologies: my previous analysis was flawed.

It turns out that a bug in the library silently overwrote the `clip` method by a value. However, `clip` isn't a settable value, it's a method. By assigning a value to it instead of invoking it, it replaced the method by that value, making it impossible to be used as a method in my subsequent javascript test, which made me believe the javascript code was drawing the exact same output.

The error is resolved with version 2.6.0, which should become available within the next 10 minutes.

I also added a demo of the now correct behavior to the demo website in the "DrawingPaths" section which replicates the demo on https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clip

BR
Stefan

**Comment by bfmsoft** [comment]
Great!  It works fine now.  Thanks for looking at it and thanks for a great package.
